### PR TITLE
Specify lowercase-hex encoding for standard algorithms only.

### PR DESCRIPTION
**Pull request by TomHennen**
There are cases where custom digests might be better expressed using different encodings.

refs #130

### CODE REVIEW

**Comment by TomHennen** [comment]
@MarkLodato I'd be interested in your thoughts here.

File: spec/v1.0-draft/digest_set.md
Comment (MarkLodato) [inline spec/v1.0-draft/digest_set.md:56]:
```suggestion
> New algorithms MUST document how the value is encoded, e.g. URL-safe base64,
```

File: spec/v1.0-draft/digest_set.md
Comment (MarkLodato) [inline spec/v1.0-draft/digest_set.md:21]: Might help to be more explicit about encoding, e.g.

```suggestion
a string encoding of the digest using that algorithm. The named standard algorithms below use lowercase hex encoding. Usually there is just a single key/value pair, but
```

(I didn't wrap this properly)

**Review by MarkLodato — COMMENTED** [review]
Sounds good to me. A few suggestions

File: spec/v1.0-draft/digest_set.md
Comment (marcelamelara) [inline spec/v1.0-draft/digest_set.md:34]: Since the `Hash1` function outputs a base64-encoded string, should we allow base64 encoding for this algorithm?

File: spec/v1.0-draft/digest_set.md
Comment (marcelamelara) [inline spec/v1.0-draft/digest_set.md:56]: Just an idea (and maybe this has been discussed already): Do we see a downside to adding an optional `encoding` field to the DigestSet type?

File: spec/v1.0-draft/digest_set.md
Comment (TomHennen) [reply spec/v1.0-draft/digest_set.md:34]: That would make sense, I wonder if it would have any backwards compatibility issues?  E.g. we'd allow EITHER lowercase-hex or base64 and users would need to support both.

@MarkLodato WDYT?

File: spec/v1.0-draft/digest_set.md
Comment (TomHennen) [reply spec/v1.0-draft/digest_set.md]: It's a good thing to discuss.  My first concern is that it would probably be a breaking change?  This would go from map<string, string> to map<string, ValueEntry> where ValueEntry has (encoding, value).  or is there a more clever way to do it?

File: spec/v1.0-draft/digest_set.md
Comment (TomHennen) [reply spec/v1.0-draft/digest_set.md]: Done

File: spec/v1.0-draft/digest_set.md
Comment (MarkLodato) [reply spec/v1.0-draft/digest_set.md:36]: That seems like it makes the producer's job ever-so-slightly easier but the consumer's job much harder, especially if most of the time they receive one encoding but then they suddenly get a different one. I'm inclined to pick one.

Between base64 and hex, I don't feel too strongly.

Comment (TomHennen) [reply spec/v1.0-draft/digest_set.md:36]: I think the best approach might be

1. Leave goModuleH1 as lowercase hex so that existing use cases can stay the same and we can unblock 1.0 with minimal risk.
2. In the future, if needed, add goModuleH1+base64 (per @marcelamelara's suggestion) so that people can specify _both_ goModuleH1 (lowercase hex) and goModuleH1+base64.

File: spec/v1.0-draft/digest_set.md
Comment (TomHennen) [reply spec/v1.0-draft/digest_set.md]: Right, I think people will generally be using the key field as a opaque string, so adding +base64 here just turns it into a new string type (which is still useful!).  Folks will be able to use both goModuleH1 and goModuleH1+base64 if they want.  Maybe in the future we could add fancy support for specifying the encoding in here, but then people would need to handle them as non-opaque strings, which would be a big change.

File: spec/v1.0-draft/digest_set.md
Comment (marcelamelara) [reply spec/v1.0-draft/digest_set.md:36]: Might be good to add a comment somewhere in the docs, or open an issue for this, just so we don't forget.

Comment (trishankatdatadog) [reply spec/v1.0-draft/digest_set.md:36]: Agree with Mark: generally, for security reasons (e.g., for the same reasons DSSE was formulated), best to decode using one expected format

**Review by joshuagl — APPROVED** [review]

File: spec/v1.0-draft/digest_set.md
Comment (marcelamelara) [inline spec/v1.0-draft/digest_set.md:30]: One more thing. Might be good to clarify the naming of the algorithms per https://github.com/in-toto/attestation/pull/135#discussion_r1114675451

File: spec/v1.0-draft/digest_set.md
Comment (TomHennen) [reply spec/v1.0-draft/digest_set.md:37]: Ok, I'll leave it as is, without a note until we actually need something else.

Thanks everyone!

File: spec/v1.0-draft/digest_set.md
Comment (TomHennen) [reply spec/v1.0-draft/digest_set.md]: Done!

**Comment by TomHennen** [comment]
I think we're good now, but I'll let this sit for a day before merging.

File: spec/v1.0-draft/digest_set.md
Comment (MarkLodato) [inline spec/v1.0-draft/digest_set.md:29]:
```suggestion
    (converting to lowercase and replacing `-` with `_`) as keys
```

**Review by MarkLodato — APPROVED** [review]

**Review by trishankatdatadog — APPROVED** [review]

**Comment by TomHennen** [comment]
Thanks everyone!

**Pull request merged**
